tracer/src/Datadog.Trace/IAST/Aspects/System/StringAspects.cs

```diff
@@ -21,6 +21,136 @@ public partial class StringAspects
 
+    /// <summary>
+    /// String.Trim aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <returns> String.Trim() </returns>
+    [AspectMethodReplace("System.String::Trim()", AspectFilter.StringLiteral_0)]
+    public static string Trim(string target)
+    {
+        return StringModuleImpl.OnStringTrim(target, target.Trim(), null, true, true);
+    }
+
+    /// <summary>
+    /// String.Trim aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <param name="trimChars"> chars to trim </param>
+    /// <returns> String.Trim() </returns>
+    [AspectMethodReplace("System.String::Trim(System.Char[])", AspectFilter.StringLiteral_0)]
+    public static string Trim(string target, char[] trimChars)
+    {
+        if (trimChars != null && trimChars.Length > 0)
+        {
+            return StringModuleImpl.OnStringTrimArray(target, target.Trim(trimChars), trimChars, true, true);
+        }
+        else
+        {
+            return StringModuleImpl.OnStringTrim(target, target.Trim(trimChars), null, true, true);
+        }
+    }
+
+    /// <summary>
+    /// String.Trim aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <param name="trimChar"> char to trim </param>
+    /// <returns> String.Trim() </returns>
+    [AspectMethodReplace("System.String::Trim(System.Char)", AspectFilter.StringLiteral_0)]
+    public static string Trim(string target, char trimChar)
+    {
+        return StringModuleImpl.OnStringTrim(target, target.Trim(trimChar), trimChar, true, true);
+    }
+
+    /// <summary>
+    /// String.TrimStart aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <param name="trimChars"> chars to trim </param>
+    /// <returns> String.TrimStart() </returns>
+    [AspectMethodReplace("System.String::TrimStart(System.Char[])", AspectFilter.StringLiteral_0)]
+    public static string TrimStart(string target, char[] trimChars)
+    {
+        if (trimChars != null && trimChars.Length > 0)
+        {
+            return StringModuleImpl.OnStringTrimArray(target, target.TrimStart(trimChars), trimChars, true, false);
+        }
+        else
+        {
+            return StringModuleImpl.OnStringTrim(target, target.TrimStart(trimChars), null, true, false);
+        }
+    }
+
+#if !NETFRAMEWORK
+    /// <summary>
+    /// String.TrimStart aspect
+    /// </summary>
+    /// <param name="target"> string base instance </param>
+    /// <param name="trimChar"> char to trim </param>
+    /// <returns> String.TrimStart() </returns>
+    [AspectMethodReplace("System.String::TrimStart(System.Char)", AspectFilter.StringLiteral_0)]
+    public static string TrimStart(string target, char trimChar)
+    {
+        return StringModuleImpl.OnStringTrim(target, target.TrimStart(), trimChar, true, false);
```

Comment:
Should be calling target.TrimStart(trimChar) right? If yes, do you have a test for this one? 

```suggestion
        return StringModuleImpl.OnStringTrim(target, target.TrimStart(trimChar), trimChar, true, false);
```

Reply:
That's right. . Actually, there was a test for this case, but it was sending the default char in the test, so the test was not failing. I have refactored the tests and grouped them by method, reviewed them and added that case. Thank you for spotting it.